lib/_stream_writable.js

```diff
@@ -155,11 +155,10 @@ Writable.prototype.pipe = function() {
 };
 
 
-function writeAfterEnd(stream, cb) {
-  var er = new Error('write after end');
-  // TODO: defer error events consistently everywhere, not just the cb
-  stream.emit('error', er);
-  process.nextTick(cb, er);
+function writeAfterEndErr(stream, cb) {
+  const err = new Error('write after end');
+  process.nextTick(() => stream.emit('error', err));
```

Comment:
> It would probably be better to just call process.nextTick once here and both emit the error and call the callback.

Sure thing, agreed it looked strange. Just pushed an update.